### PR TITLE
Fix windows build and add github CI

### PR DESCRIPTION
**Pull request by jschwe**
See the individual commits for details

### CODE REVIEW

**Comment by jschwe** [comment]
@mrobinson I based the github action on your previous MR, but cleaned it up slightly. The github CI passes on my fork: https://github.com/jschwe/fontsan/actions/runs/7680490707

I also fixed some other minor things I stumbled across. Probably it would also be a good idea to update the dependencies to a newer version, but I thought it would be best to get CI running again before attempting to update ots.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: fontsan CI
```
Comment (mrobinson) [inline .github/workflows/test.yml:1]: I'm not sure it's necessary to repeat `fontsan` here.

```suggestion
name: CI
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+on: push
```
Comment (mrobinson) [inline .github/workflows/test.yml:3]: Can you also enable this workflow for pull requests?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    runs-on: ubuntu-latest
+    needs:
+      - "build_and_test"
+    steps:
+      - name: Mark the job as successful
+        run: exit 0
+        if: success()
+      - name: Mark the job as unsuccessful
+        run: exit 1
+        if: "!success()"
```
Comment (mrobinson) [inline .github/workflows/test.yml:43]: These should be modified to take into account all task dependencies. See https://github.com/servo/webrender/blob/master/.github/workflows/main.yml#L53. This is important for the proper functioning of the merge queue.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Cargo build
+        run: cargo build -vv
+
+      - name: Cargo test
+        run: cargo test -vv
+        env:
+          RUST_BACKTRACE: 1
+
+  build_result:
+    name: homu build finished
```
Comment (mrobinson) [inline .github/workflows/test.yml:33]: This repository no longer uses homu.

```suggestion
    name: Result
```

**Review by mrobinson — CHANGES_REQUESTED** [review]
Nicely done. I have a few small suggestions.

**Comment by jschwe** [comment]
Thanks for the review - I amended the `github CI workflow` commit to include all your suggestions about the workflow

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          toolchain: stable
+
+      - name: Cargo build
+        run: cargo build -vv
+
+      - name: Cargo test
+        run: cargo test -vv
+        env:
+          RUST_BACKTRACE: 1
```
Comment (mrobinson) [inline .github/workflows/test.yml:30]:
```suggestion
      - uses: actions/checkout@v4
        with:
          submodules: recursive
      - uses: dtolnay/rust-toolchain@master
        with:
          toolchain: stable
      - name: Build
        run: cargo build -vv
      - name: Test
        run: cargo test -vv
        env:
          RUST_BACKTRACE: 1
```

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
@jschwe Thanks for your contribution!

**Pull request merged**